### DISCUSSION TITLE
Stable and predictable release cadence and version numbers

### DISCUSSION THREAD

**Discussion by EwoutH** [openthread/openthread#8239] [General]
Currently OpenThread doesn't have clear version tags or releases, and it isn't clear when a new release would be made. A stable, predictable release cadence with logical version numbers (for example [Semantic Versioning](https://semver.org/)) would make adopting and maintaining OpenThread implementations easier.

**Comment by jwhui** [comment]
The OpenThread project so far has avoided making formal releases and maintaining release branches. While this may not be conventional software engineering practice, this has not been an issue for integrators (e.g. those maintaining IEEE 802.15.4 chip vendor SDKs) so far. We have fairly extensive Continuous Integration testing, which attempts to keep main branch near or at production quality at all times. An integrator can also choose to maintain their own stable branches of OpenThread.

I'm hoping to make the specific commits that were used to achieve Thread Certification by users of OpenThread more visible, which should help.
